core/apple/test/NSInputStreamSourceTest.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENCE file.
+ */
+
+package kotlinx.io
+
+import platform.Foundation.NSInputStream
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.fail
+
+class NSInputStreamSourceTest {
+    @Test
+    fun nsInputStreamSource() {
+        val input = NSInputStream(byteArrayOf(0x61).toNSData())
```

Comment:
Different kinds of NSInputStreams (files, in-memory, etc) have different implementations, its worth checking that we correctly handle all of them, not only array-backed on.

I guess it may not be that easy to check socket-based one, but for files it should be relatively simple.

Reply:
I added this `NSInputStream` from file URL test: https://github.com/Kotlin/kotlinx-io/pull/174/commits/002141238f5a3ac19f309bb47436160f6912cea9

I'm confused why this works though, as the `file` path string is always 1 or 2 bytes of seemingly garbage (not characters), but somehow it creates a `NSInputStream` from this file `NSURL` and reads what the `Sink` wrote to the file. 🤔

Reply:
I guess it is just creating a file with the junk character(s) name in the current working directory. Probably need to revisit `createTempFile()` for Apple platforms. `mktemp()` doesn't always return null, but it doesn't seem to ever return something valid.

Also, `"$tmpDir/tmp-${rnd.nextInt()}"` puts a double `//` in the path: `...//tmp-...`.

Reply:
I created https://github.com/Kotlin/kotlinx-io/issues/183 to track this.

Reply:
Would you consider replacing `createTempFile` with `"${NSTemporaryDirectory()}${NSUUID().UUIDString()}"` for now?

I don't want fixing #183 concurrently with your PR to avoid additional merge conflicts in build scripts (as native test sources should be split into appleTest and nonAppleTest) and would rather fix it after.

Reply:
I made the change and left a comment about the future fix.